BUILD

```diff
@@ -0,0 +1,54 @@
+################################# define local and nolocal #################################
+# Conditional compile definitions based on --define=local=<value> build flag
+config_setting(
+    name = "is_local",
+    values = {"define": "local=true"},
+)
+
+# Set defines based on whether LOCAL is defined
+local_defines = select({
+    "//conditions:default": ["NOLOCAL=1"],  
+    ":is_local": ["LOCAL=1"],               
+})
+
+################################# library BUILD target (files in src and include) #################################
+cc_library(
+    name = "include_files",
+    hdrs = glob(["include/**/*.h"]),
+    deps = ["@com_github_nlohmann_json//:json"],
+    visibility = ["//visibility:public"],
+)
+
+cc_library(
+    name = "src_files",
+    srcs = glob(["src/**/*.cpp"]),
+    hdrs = glob(["src/**/*.h"]),
+    visibility = ["//visibility:public"],
+    deps = [":include_files"],
+    defines = local_defines, 
+)
+
+################################# main BUILD target #################################
+cc_binary(
+    name = "bootstrap",
+    srcs = ["main.cpp"],
+    copts = ["-std=c++20", "-Wall", "-pthread"],
```

Comment:
When trying to bazel build this project on linux, I ran into the following issue:
```
INFO: Analyzed 5 targets (0 packages loaded, 0 targets configured).
ERROR: /home/jordan/PeerToPeer/BUILD:40:10: Linking provider failed: (Exit 1): gcc failed: error executing CppLink command (from target //:provider) /usr/bin/gcc @bazel-out/k8-fastbuild/bin/provider-2.params

Use --sandbox_debug to see verbose messages from the sandbox and retain the sandbox build root for debugging
bazel-out/k8-fastbuild/bin/_objs/src_files/provider.pic.o:provider.cpp:function std::thread::thread<void (Provider::*)(), Provider*, void>(void (Provider::*&&)(), Provider*&&): error: undefined reference to 'pthread_create'
collect2: error: ld returned 1 exit status
Use --verbose_failures to see the command lines of failed build steps.
INFO: Elapsed time: 0.636s, Critical Path: 0.09s
INFO: 2 processes: 2 internal.
ERROR: Build did NOT complete successfully
```

I was able to resolve this issue by adding the following line to each cc_binary:
```
linkopts = ["-lpthread"],
```

Reply:
added in the .bazelrc file